test/integration/services/mapping/goodsshipment/GoodsShipmentBuilderIntegrationSpec.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package integration.services.mapping.goodsshipment
+
+import forms.declaration.ConsigneeDetailsSpec.correctConsigneeDetailsFull
+import forms.declaration.ConsignmentReferencesSpec.correctConsignmentReferences
+import forms.declaration.DeclarationAdditionalActorsSpec.{correctAdditionalActors1, correctAdditionalActors2}
+import forms.declaration.DestinationCountriesSpec.correctDestinationCountries
+import forms.declaration.DocumentSpec.correctPreviousDocument
+import forms.declaration.GoodsLocationTestData.correctGoodsLocation
+import forms.declaration.NatureOfTransactionSpec.correctNatureOfTransaction
+import forms.declaration.PreviousDocumentsData
+import forms.declaration.WarehouseIdentificationSpec.correctWarehouseIdentification
+import models.declaration.{DeclarationAdditionalActorsData, Locations}
+import org.scalatest.{Matchers, WordSpec}
+import services.cache.{CacheTestData, ExportsCacheModelBuilder}
+import services.mapping.goodsshipment.{ConsigneeBuilder, GoodsShipmentBuilder, GoodsShipmentNatureOfTransactionBuilder}
+import wco.datamodel.wco.dec_dms._2.Declaration
+
+class GoodsShipmentBuilderIntegrationSpec
+    extends WordSpec with Matchers with CacheTestData with ExportsCacheModelBuilder {
+
+  private def builder = new GoodsShipmentBuilder(new GoodsShipmentNatureOfTransactionBuilder(), new ConsigneeBuilder())
+
+  "GoodsShipmentBuilder" should {
+
+    "correctly map to the WCO-DEC GoodsShipment instance" in {
+
+      val cacheModel = aCacheModel(
+        withNatureOfTransaction(correctNatureOfTransaction.natureType),
+        withParties(
+          createEmptyParties().copy(
+            consigneeDetails = Some(correctConsigneeDetailsFull),
+            declarationAdditionalActorsData =
+              Some(DeclarationAdditionalActorsData(Seq(correctAdditionalActors1, correctAdditionalActors2)))
+          )
+        ),
+        withLocations(
+          Locations().copy(
+            goodsLocation = Some(correctGoodsLocation),
+            destinationCountries = Some(correctDestinationCountries),
+            warehouseIdentification = Some(correctWarehouseIdentification)
```

Comment:
Instead of `withLocations` you can do 
```
withGoodsLocation(...),
withDestinationCountries(...),
withWarehouseIdentification(...)`
```
I'm pretty sure (most of) those already exist and I think it reads better without all the `copy`s which kind of defeat the point of the builder

Reply:
:thinking:

Reply:
ok will change to match this.